src/native/libs/System.Native/entrypoints.c

```diff
@@ -34,6 +34,7 @@
 static const Entry s_sysNative[] =
 {
     DllImportEntry(SystemNative_FStat)
+#if !defined(TARGET_APPLE) || defined(TARGET_OSX)
```

Comment:
It seems that we tend to keep this list the same for all platforms and rather handle the missing functionality in the actual implementation of the methods. E.g. for WASI, many of the methods return EINVAL or NULL.

Reply:
This is preexisting implementation where it was not done that way. We just never compiled the entrypoints.c on mobile platforms.

Since this is used for corehost which is not particularly useful on mobile platforms maybe I can just drop this part of change and try to exclude corehost build.

Reply:
Done. There are some additional changes I can revert once https://github.com/dotnet/runtime/pull/110092 and https://github.com/dotnet/runtime/pull/110023 gets in.

Reply:
Rebased after https://github.com/dotnet/runtime/pull/110092 merge now.